### PR TITLE
feat: add hypergeometric-entropy 

### PR DESCRIPTION
**Pull request by 24archit**
Resolves N.A

## Description

> What is the purpose of this pull request?

This pull request:

-   Implements the **differential entropy** calculation for the **hypergeometric distribution** (`@stdlib/stats/base/dists/hypergeometric/entropy`).
-   Utilizes **robust Kahan-Babuška summation** (`@stdlib/stats/incr/sum`) to ensure numerical stability and accuracy.
-   Adds comprehensive **unit tests** covering edge cases, parameter validation, and correctness checks against known values.
-   Adds **TypeScript definitions** (`index.d.ts`, `test.ts`) and **benchmarks** (`benchmark/benchmark.js`).
-   Includes complete **documentation** with runnable examples and REPL help text.
-   Updates the parent package (`@stdlib/stats/base/dists/hypergeometric`) to export the new `entropy` function.

## Related Issues

> Does this pull request have any related issues?

This pull request has the following related issues:

-  N.A

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

> When authoring the changes proposed in this PR, did you use any kind of AI assistance?

-   [] Yes
-   [x] No

If you answered "yes" above, how did you use AI assistance?

-   [] Code generation (e.g., when writing an implementation or fixing a bug)
-   [] Test/benchmark generation
-   [] Documentation (including examples)
-   [] Research and understanding

#### Disclosure

> If you answered "yes" to using AI assistance, please provide a short disclosure indicating how you used AI assistance. This helps reviewers determine how much scrutiny to apply when reviewing your contribution. Example disclosures: "This PR was written primarily by Claude Code." or "I consulted ChatGPT to understand the codebase, but the proposed changes were fully authored manually by myself.".

N.A

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by 24archit** [comment]
Hi, @kgryte  and @Planeshifter , just checking to see if any further updates are needed from my end ? Thanks!

**Comment by 24archit** [comment]
Hi team, just bumping this gently as the workflows are still pending approval for a first-time contributor. I’ve verified the tests pass locally. Thanks!

File: lib/node_modules/@stdlib/stats/base/dists/hypergeometric/entropy/README.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+// returns NaN
+
+y = entropy( 20, NaN, 4 );
+// returns NaN
+
+y = entropy( 20, 10, NaN );
+// returns NaN
+```
+
+If provided a population size `N`, subpopulation size `K` or draws `n` which is not a nonnegative integer, the function returns `NaN`.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/hypergeometric/entropy/README.md]: Missing the Oxford comma after `` `K` ``. The kurtosis README has: "subpopulation size `K`**,** or draws `n`".

```suggestion
If provided a population size `N`, subpopulation size `K`, or draws `n` which is not a nonnegative integer, the function returns `NaN`.
```

File: lib/node_modules/@stdlib/stats/base/dists/hypergeometric/entropy/README.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+// returns NaN
+
+y = entropy( 10, 1.5, 2 );
+// returns NaN
+
+y = entropy( 10, 5, -2.0 );
+// returns NaN
+```
+
+If the number of draws `n` or subpopulation size `K` exceeds population size `N`, the function returns `NaN`.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/hypergeometric/entropy/README.md]: Two small wording deviations from the kurtosis reference: missing "the" before "subpopulation size", and "exceeds" should be "exceed" (plural, since it's "draws `n` or ... `K`" as compound subject).

```suggestion
If the number of draws `n` or the subpopulation size `K` exceed population size `N`, the function returns `NaN`.
```

**Comment by 24archit** [comment]
Hi team, checking in one more time! No rush on the full code review, but whenever someone has a chance to approve the workflow runs for this PR, that would be awesome so we can get the CI checks out of the way. Everything is passing on my local machine. Thank you!
